src/goto-programs/goto_convert.cpp

```diff
@@ -1218,7 +1187,7 @@ exprt goto_convertt::case_guard(
     dest.move_to_operands(eq_expr);
   }
 
-  assert(!dest.operands().empty());
+  CHECK_RETURN(!dest.operands().empty());
```

Comment:
This code needs review. It should just use `disjunction` to avoid the `if(dest.operands().size()==1)` below. Furthermore the `assert`/`CHECK_RETURN` should be lifted further up as this invariant holds if, and only if, `!case_op.empty()`.

Reply:
Should arguably be `PRECONDITION(!case_op.operands.empty())` + `INVARIANT(dest.operands().size() == case_op.operands.size())`.